tools/autograd/gen_autograd.py

```diff
@@ -42,21 +42,28 @@
     'as_strided': 'self',
     'diagonal': 'self',
     'expand': 'self',
+    'expand_as': 'self',
     'narrow': 'self',
     'permute': 'self',
+    'reshape': 'self',
+    'reshape_as': 'self',
     'select': 'self',
     'slice': 'self',
     'squeeze': 'self',
     't': 'self',
     'transpose': 'self',
     'unfold': 'self',
     'unsqueeze': 'self',
+    'flatten': 'flatten',
     'view': 'self',
+    'view_as': 'self',
     'unbind': 'self',
     '_indices': 'self',
     '_values': 'self',
     'indices': 'self',
     'values': 'self',
+    'contiguous': 'self',
+    'pin_memory': 'pin_memory',
```

Comment:
This one does not always returns a view. So it should not be here I'm afraid.
I think we want to make it composite and add here only the function corresponding to the case where it is actually a view.

Reply:
It's not clear if that means `pin_memory` should be in the second list or omitted entirely.